### DISCUSSION TITLE
Restoring pg_control file

### DISCUSSION THREAD

**Discussion by sh-soltanpour** [pgmoneta/pgmoneta#206] [General]
Hi everyone,
I'm working on a project about restoring pg_control in the restore process, and I thought it would be useful to ask my questions publicly here.

In the project requirement, I am told to skip restoring the `pg_control` file in the `first round` and restore it explicitly in the `second round`. My questions are:

1- What's the benefit of skipping the `pg_control`  file in the first round?
2- By round, do you mean the workflows in the code? What I have figured out is that we can create another workflow that is executed after the `pgmoneta_workflow_create_restore`. For example, named `pgmoneta_workflow_restore_pg_control` which restores pg_control explicitly. I just wanted to make sure that my understanding from the first round and second round is correct. If so, I'll go ahead and implement this additional workflow. 

Thanks,
Shahryar

**Comment by jesperpedersen** [comment]
Well, the files are copied, decrypted and uncompressed.

The problem is that once `pg_control` is "ready" then the database cluster can be started - even though some files are still being uncompressed f.ex...

I would look into a solution where an array of special file names are passed and skipped, and then processed after all other files have gone through the existing pipeline.

It is ok for the array of file names to be global...

https://github.com/pgmoneta/pgmoneta/blob/main/src/libpgmoneta/workflow.c#L169 is a good starting point

**Reply by sh-soltanpour** [reply]
Makes sense!
Thanks a lot for helpful information.
